# system error code
uint32 NONE = 0
#   // warning
uint32 SYSTEM_ERROR_CODE_BATTERY_LOW_WARN = 1
uint32 SYSTEM_ERROR_CODE_MOTOR_OVERHEAT_WARN = 2
uint32 SYSTEM_ERROR_CODE_MOTOR_OVERCURRENT_WARN = 4
uint32 SYSTEM_ERROR_CODE_WARN_BIT_MASK = 4095
#   // fault
uint32 SYSTEM_ERROR_CODE_BATTERY_LOW_FAULT = 65536
uint32 SYSTEM_ERROR_CODE_MOTOR_OVERHEAT_FAULT = 131072
uint32 SYSTEM_ERROR_CODE_MOTOR_OVERCURRENT_FAULT = 262144
uint32 SYSTEM_ERROR_CODE_MOTOR_COMM_FAULT = 524288
uint32 SYSTEM_ERROR_CODE_FAULT_BIT_MASK = 268369920

# system oper state
uint32  SYSTEM_OPERATIONAL_STATE_OPERATIONAL = 0
uint32  SYSTEM_OPERATIONAL_STATE_INITIALIZATION = 1
uint32  SYSTEM_OPERATIONAL_STATE_MAINTENANCE = 2
uint32  SYSTEM_OPERATIONAL_STATE_SOFTWARE_UPDATE = 3
uint32  SYSTEM_OPERATIONAL_STATE_ESTOP_ACTIVATED = 4
uint32  SYSTEM_OPERATIONAL_STATE_HARDWARE_FAULT = 5

# system control state
uint32  SYSTEM_CONTROL_STATE_UNINITIALIZED = 0
uint32  SYSTEM_CONTROL_STATE_STANDBY = 1
uint32  SYSTEM_CONTROL_STATE_RC_HALT_TRIGGERED = 2
uint32  SYSTEM_CONTROL_STATE_RC_MANUAL_CONTROL = 3
uint32  SYSTEM_CONTROL_STATE_CAN_COMMAND_CONTROL = 4

bool rc_connected
uint32  error_code
uint32  operational_state
uint32  control_state
